librz/arch/p/asm/asm_pic.c

```diff
@@ -5,26 +5,26 @@
 #include <rz_asm.h>
 #include <rz_lib.h>
 
-#include "pic/pic_baseline.h"
-#include "pic/pic_pic18.h"
-#include "pic/pic_midrange.h"
+#include "pic/pic14.h"
+#include "pic/pic18.h"
+#include "pic/pic16.h"
 
 static int asm_pic_disassemble(RzAsm *a, RzAsmOp *op, const ut8 *b, int l) {
 	int res = -1;
-	if (a->cpu && strcasecmp(a->cpu, "baseline") == 0) {
-		res = pic_baseline_disassemble(op, b, l);
-	} else if (a->cpu && strcasecmp(a->cpu, "midrange") == 0) {
-		res = pic_midrange_disassemble(op, b, l);
-	} else if (a->cpu && strcasecmp(a->cpu, "pic18") == 0) {
-		res = pic_pic18_disassemble(op, b, l);
+	if (a->cpu && is_pic14(a->cpu)) {
+		res = pic14_disassemble(a, op, b, l);
+	} else if (a->cpu && is_pic16(a->cpu)) {
+		res = pic16_disassemble(a, op, b, l);
+	} else if (a->cpu && is_pic18(a->cpu)) {
+		res = pic18_disassemble(a, op, b, l);
 	}
 	return op->size = res;
 }
 
 RzAsmPlugin rz_asm_plugin_pic = {
 	.name = "pic",
 	.arch = "pic",
-	.cpus = "baseline,midrange,pic18",
+	.cpus = "pic18,pic16,pic14,pic12,pic10,highend,midrange,baseline",
```

Comment:
I see that `pic12` and `pic10` aren't handled in the above function.

Reply:
yes